Service/AdditionalSection/IndexBuilder.php

```diff
@@ -66,25 +75,23 @@ public function buildIndex(int $storeId, ?array $entityIds, ?array $options): vo
 
             $indexName = $this->additionalSectionHelper->getIndexName($storeId);
             $indexName = $indexName . '_' . $section['name'];
+            $tempIndexName = $indexName . IndexNameFetcher::INDEX_TEMP_SUFFIX;
 
             $attributeValues = $this->additionalSectionHelper->getAttributeValues($storeId, $section);
 
-            $tempIndexName = $indexName . IndexNameFetcher::INDEX_TEMP_SUFFIX;
+            $indexOptions = $this->indexOptionsBuilder->buildWithEnforcedIndex($indexName, $storeId);
```

Comment:
I think this code would also benefit from an implementation of `EntityIndexOptionsBuilderInterface`.